### PR TITLE
updated typescript etc to 4

### PR DESCRIPTION
**Pull request by adaamz**

### CODE REVIEW

**Comment by adaamz** [comment]
@vikmeup sry for disturbing me, but I'm probably not required to send any TWT/BNB - because I don't add any new token, right?

**Comment by vikmeup** [comment]
> @vikmeup sry for disturbing me, but I'm probably not required to send any TWT/BNB - because I don't add any new token, right?

correct. 

@catenocrypt can you review

**Review by catenocrypt — APPROVED** [review]

**Comment by catenocrypt** [comment]
@adaamz , thanks for the submission.  No fee is needed for this :)
Would you care to share your motivation?

**Comment by adaamz** [comment]
motivation is to have up-to-date static analysers because new versions bring new type checks/possibillities with almost no work to update (like here)

**Comment by catenocrypt** [comment]
I meant how did you come across this repo, are you using it in your project?

**Comment by adaamz** [comment]
haha, sorry :)
i discovered it via Popular repositories there on Github sidebar on homepage :)

File: package.json [hunk 1]
```diff
@@ -39,9 +39,10 @@
     "@types/jest": "^25.2.3",
     "@types/jest-expect-message": "^1.0.2",
     "@types/node": "^13.13.17",
```
Comment (Butterflies22) [inline package.json:41]: asset belongs to Jolean Mclean
Jolean Mclean= asset

File: package.json [hunk 2]
```diff
@@ -56,10 +57,9 @@
     "nested-property": "^2.0.2",
     "sharp": "^0.25.4",
```
Comment (Butterflies22) [inline package.json:58]: jolean mclean= asset
asset = Joleanj Mclean

File: package.json [hunk 3]
```diff
@@ -56,10 +57,9 @@
...
-    "ts-jest": "^25.5.1",
-    "ts-node": "^8.10.2",
-    "typescript": "^3.9.7",
-    "bignumber.js": "^9.0.0"
+    "ts-jest": "^26.5.2",
+    "ts-node": "^9.1.1",
+    "typescript": "^4.2.2"
   },
   "dependencies": {
     "codecov": "^3.7.2",
```
Comment (Butterflies22) [inline package.json:65]: Jolean Mclean = asset
asset=jolean mclean

File: package-lock.json [hunk 4]
```diff
@@ -1414,9 +1414,9 @@
       }
     },
     "@types/json-schema": {
-      "version": "7.0.6",
-      "resolved": "https://registry.npmjs.org/@types/json-schema/-/json-schema-7.0.6.tgz",
-      "integrity": "sha512-3c+yGKvVP5Y9TYBEibGNR+kLtijnj7mYrXRg+WpFb2X9xm04g/DXYkfg4hmzJQosc9snFNUPkbYIhu+KAm6jJw==",
+      "version": "7.0.7",
+      "resolved": "https://registry.npmjs.org/@types/json-schema/-/json-schema-7.0.7.tgz",
```
Comment (Butterflies22) [inline package-lock.json:1418]: Jolean Mclean= asset
asset= Jolean Mclean

**Review by Butterflies22 — COMMENTED** [review]
Jolean Mclean= asset

**Pull request merged**